packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -833,26 +862,43 @@ contract NounsDAOLogicV2 is NounsDAOStorageV2, NounsDAOEventsV2 {
     }
 
     /**
-     * @notice Changes vetoer address
-     * @dev Vetoer function for updating vetoer address
+     * @notice Begins transition of vetoer. The newPendingVetoer must call _acceptVetoer to finalize the transfer.
+     * @param newPendingVetoer New Pending Vetoer
      */
-    function _setVetoer(address newVetoer) public {
-        require(msg.sender == vetoer, 'NounsDAO::_setVetoer: vetoer only');
+    function _setPendingVetoer(address newPendingVetoer) public {
+        if (msg.sender != vetoer) {
+            revert VetoerOnly();
+        }
+
+        emit NewPendingVetoer(pendingVetoer, newPendingVetoer);
+
+        pendingVetoer = newPendingVetoer;
+    }
 
-        emit NewVetoer(vetoer, newVetoer);
+    function _acceptVetoer() external {
+        if (msg.sender != pendingVetoer || pendingVetoer == address(0)) {
```

Comment:
`pendingVetoer == address(0)` will never be evaluated, right? If `pendingVetoer` is `address(0)`, `msg.sender` will never be `pendingVetoer`.

Reply:
@solimander yes I agree. I was mostly following the same pattern as in [_acceptAdmin](https://github.com/nounsDAO/nouns-monorepo/blob/verbs-nip-1-save-proposal-creation-block/packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol#L819)
but I agree it's probably redundant.
do you see any reason for the `msg.sender != address(0)` in the `_acceptAdmin` function?

Reply:
Nope, it's redundant in both cases. Only protects against the unfeasible scenario in which the private key to the zero address is found.

Reply:
OK, removed it from _acceptVetoer: https://github.com/nounsDAO/nouns-monorepo/pull/554/commits/3a36e0ae6ffebe3c415f3f20c70988c97575901f